services/user_service/database_connector.py

```diff
@@ -0,0 +1,64 @@
+import json
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import db
+
+#implements CRUD operations interface
+class DatabaseAdmin:
+
+    def __init__(self,databaseURL, credentials_path):
+        cred = credentials.Certificate(credentials_path)
+        firebase_admin.initialize_app(cred,
+        {
+            'databaseURL' : databaseURL,
+        })
+        self.__users_ref = db.reference("users")
+
+
+    def get_all_users(self):
+        return self.__users_ref.get()
+    
+
+    def get_user_by_email(self, email):
+        for user_id in self.get_user_ids():
+            if email == self.__users_ref.child(user_id).child("email").get():
+                return self.__users_ref.child(user_id).get()
+    
+
+    def get_user_ids(self):
+        ids = [i for i in self.__users_ref.get()]
+        return ids
```

Comment:
??
Seems like the same as self.users_ref.get() (so the same as get_all_users())

Reply:
Can you please explain the difference between this and get_all_users() before resolving conversation?

Reply:
get_all_users returns json object with all the data related to the user, when get_user_ids() returns only list of ids